src/selectors/WalletSelectors.ts

```diff
@@ -41,42 +42,68 @@ export const getActiveWalletCurrencyInfos = (
 }
 
 export const getExchangeRate = (
-  state: RootState,
-  fromCurrencyCode: string,
+  exchangeRates: GuiExchangeRates,
+  pluginId: string,
+  tokenId: EdgeTokenId,
   toCurrencyCode: string
 ): number => {
-  const exchangeRates = state.exchangeRates
-  const rateKey = `${fromCurrencyCode}_${toCurrencyCode}`
-  const rate = exchangeRates[rateKey] ?? 0
-  return rate
+  const rateObj =
+    exchangeRates.crypto[pluginId]?.[tokenId ?? '']?.[toCurrencyCode]
+  if (rateObj?.current != null) return rateObj.current
+
+  // if not found, try to find USD path
+  const rateUSD =
+    exchangeRates.crypto?.[pluginId]?.[tokenId ?? '']?.['iso:USD']?.current ?? 0
+  const fiatUSD =
+    exchangeRates.fiat?.[toCurrencyCode]?.['iso:USD']?.current ?? 0
+  if (rateUSD === 0 || fiatUSD === 0) return 0
+
+  const foundRate = rateUSD / fiatUSD
+  return foundRate
+}
+
+export const getFiatExchangeRate = (
+  state: RootState,
+  fromIsoCode: string,
+  toIsoCode: string
+): number => {
+  const rate = state.exchangeRates.fiat[fromIsoCode]?.[toIsoCode]
+  if (rate?.current != null) return rate.current
+
+  const fromUSD = state.exchangeRates.fiat?.[fromIsoCode]?.['iso:USD']?.current
+  const toUSD = state.exchangeRates.fiat?.[toIsoCode]?.['iso:USD']?.current
+  if (fromUSD === 0 || toUSD === 0) return 0
+
+  const foundRate = fromUSD / toUSD
+  return foundRate
 }
 
 export const convertCurrency = (
   state: RootState,
-  fromCurrencyCode: string,
+  pluginId: string,
+  tokenId: EdgeTokenId,
   toCurrencyCode: string,
   amount: string = '1'
 ): string => {
-  const exchangeRate = getExchangeRate(state, fromCurrencyCode, toCurrencyCode)
+  const exchangeRate = getExchangeRate(
```

Comment:
This is literally the same as `convertCurrencyFromExchangeRates`, but with an extra `state.`. Do we really need both functions?